### PR TITLE
Fix composer test failures due to invalid lock

### PR DESCRIPTION
**Pull request by mkaz**
## Description

The `dealerdirect/phpcodesniffer-composer-installer` package needed to be upgraded to 0.7 to support composer 2.

## How has this been tested?

Run: `npm run wp-env run composer 'install --no-interaction'` and confirm error, you can also see this in recent continuous integration test failures.

```
The "dealerdirect/phpcodesniffer-composer-installer" plugin was skipped because it requires a Plugin API version ("^1.0") that does not match your Composer installation ("2.0.0"). You may need to run composer update with the "--no-plugins" option.
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Your lock file does not contain a compatible set of packages. Please run composer update.
```

Apply the change, and run and the error is cleared up. 👍

## Types of changes

- Updates composer package version
- Updates composer lock file

### CODE REVIEW

**Review by swissspidy — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -23,6 +23,6 @@
 	},
 	"scripts": {
 		"format": "phpcbf --standard=phpcs.xml.dist --report-summary --report-source",
-		"lint": "phpcs --standard=phpcs.xml.dist"
+		"lint": "phpcs --standard=phpcs.xml.dist --runtime-set ignore_warnings_on_exit 1"
```
Comment (swissspidy) [inline composer.json:26]: Why this change? Seems unrelated?

Comment (mkaz) [reply composer.json:26]: Without that change the lint warnings would cause the test to fail on 7 warnings for

`VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable`

```
Script phpcs --standard=phpcs.xml.dist handling the lint event returned with error code 1
✖ Command failed with exit code 1
```

**Comment by noahtallen** [comment]
I went ahead and merged this to unblock other PRs. Thanks for the fix!

We can always follow-up on `ignore_warnings_on_exit` if we need to. :)

**Comment by mkaz** [comment]
Thanks @noahtallen - a follow up task can be to clear the warnings and then remove the flag.
